lib/api/pokedex_api_spec/model/pokemon_model.dart

```diff
@@ -0,0 +1,16 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'pokemon_model.freezed.dart';
+
+part 'pokemon_model.g.dart';
+
+@freezed
+class Pokemon with _$Pokemon {
+  factory Pokemon({
+    @Default('') @JsonKey(name: 'name') String name,
+    @Default(1) @JsonKey(name: 'id') int id,
```

Comment:
@yohancai have you checked the api for this? As I checked the api based on your target endpoint for this, here's the response structure I saw 🙂:

![Screen Shot 2022-08-03 at 12 08 19 PM](https://user-images.githubusercontent.com/83454273/182522148-de889bdd-543e-4f9b-acef-ea11c91192ea.png)

Note: kindly realign your model based on the structure given from the api 🙂